api/v1alpha3/azurecluster_types.go

```diff
@@ -96,6 +100,16 @@ type AzureClusterList struct {
 	Items           []AzureCluster `json:"items"`
 }
 
+// GetConditions returns the list of conditions for a AzureCluster API object.
+func (c *AzureCluster) GetConditions() clusterv1.Conditions {
+	return c.Status.Conditions
+}
+
+// SetConditions will set the given conditions on an AzureCluster object
+func (c *AzureCluster) SetConditions(conditions clusterv1.Conditions) {
```

Comment:
Most getters/setters live in the scope package at the moment (`SetProviderID`, `SetReady`, `GetVMState`, etc.). Not sure if that's the best place and maybe that's something we need to discuss separately but I think it makes more sense to keep those together for now.

Reply:
ok will move them

Reply:
if I move these to scope, I can't use them in the manual conversion which is required

Reply:
gotcha, let's just leave them then. At one point we'll have to go through scope/api types and make sure all of these are where they should be and are unit tested...

Reply:
We can always add a proxy func on the scope which sets the condition on the underlying resource.